$NetBSD: patch-Makefile,v 1.1 2025/07/10 15:47:58 dkazankov Exp $

Add gprbuild and gprinstall options
Make install target for a library

--- Makefile.orig	2024-01-03 19:00:52.000000000 +0200
+++ Makefile
@@ -4,6 +4,9 @@
 BUILD_DIR ?= .
 INSTALL_DIR ?= .
 
+GPRBUILD_OPTIONS ?= -p -j$(PROCESSORS)
+GPRINSTALL_OPTIONS ?= 
+
 ALL_LIBRARY_TYPES = static static-pic relocatable
 
 .PHONY: lib
@@ -11,17 +14,29 @@
 	gprbuild -k -P adasat.gpr -p -j$(PROCESSORS) \
 		--relocate-build-tree="$(BUILD_DIR)" \
 		-XLIBRARY_TYPE=$(LIBRARY_TYPE) \
-		-XBUILD_MODE=$(BUILD_MODE)
+		-XBUILD_MODE=$(BUILD_MODE) \
+		$(GPRBUILD_OPTIONS)
 
 .PHONY: all-libs
 all-libs:
 	for kind in $(ALL_LIBRARY_TYPES) ; do \
-		gprbuild -k -P adasat.gpr -p -j$(PROCESSORS) \
+		gprbuild -k -P adasat.gpr $(GPRBUILD_OPTIONS) \
 			--relocate-build-tree="$(BUILD_DIR)" \
 			-XLIBRARY_TYPE=$$kind \
 			-XBUILD_MODE=$(BUILD_MODE) ; \
 	done
 
+.PHONY: install-lib
+install-lib:
+	gprinstall -P adasat.gpr -p -f \
+		-XLIBRARY_TYPE=$(LIBRARY_TYPE) \
+		-XBUILD_MODE=$(BUILD_MODE) \
+		--relocate-build-tree="$(BUILD_DIR)" \
+		--prefix="$(INSTALL_DIR)" \
+		--build-name=$(LIBRARY_TYPE) \
+		--build-var=LIBRARY_TYPE ; \
+		$(GPRINSTALL_OPTIONS) \
+
 .PHONY: install
 install:
 	for kind in $(ALL_LIBRARY_TYPES) ; do \
@@ -32,6 +47,7 @@
 			--prefix="$(INSTALL_DIR)" \
 			--build-name=$$kind \
 			--build-var=LIBRARY_TYPE ; \
+			$(GPRINSTALL_OPTIONS) \
 	done
 
 .PHONY: test
